workers/src/main/scala/workers/Cache.scala

```diff
@@ -0,0 +1,146 @@
+import org.scalajs.dom.IDBFactory
+import org.scalajs.dom.{Cache => JsCache}
+
+import java.time.Instant
+import java.util.Base64
+import scala.scalajs.js.typedarray.Int8Array
+
+import scalajs.js
+import scalajs.js.JSConverters.*
+
+/**
+ * Cacheable computation.
+ */
+case class Cacheable[F[_], I, O](name: String, version: Int, invoke: I => F[O])
+
+/**
+ * Generic cache interface.
+ */
+sealed trait Cache[F[_]]:
+  def eval[I: Pickler, O: Pickler](computation: Cacheable[F, I, O]): I => F[O]
+
+/**
+ * `NoCache` is used when, for some reason, a cache could not be initialized.
+ */
+case class NoCache[F[_]]() extends Cache[F]:
+  def eval[I: Pickler, O: Pickler](computation: Cacheable[F, I, O]): I => F[O] =
+    computation.invoke
+
+/**
+ * Cache backed up by an Indexed DB store.
+ */
+case class IDBCache[F[_]](
+  cacheDB: IndexedDb.Database,
+  store:   ObjectStoreDef.Sync[Pickled, Pickled]
+)(using
+  F:       Async[F]
+) extends Cache[F]:
+  override def eval[I: Pickler, O: Pickler](computation: Cacheable[F, I, O]): I => F[O] = { input =>
+    val pickledInput: Pickled = Pickled(asBytes((computation.name, computation.version, input)))
+
+    // F.delay(println(s"CACHED EVAL $input")) >>
+    cacheDB
+      .get(store)(pickledInput)
+      .toF
+      // .flatTap(pickled => F.delay(println(pickled.fold("CACHE MISS")(_ => "CACHE HIT"))))
+      .flatMap(
+        _.fold(
+          computation
+            .invoke(input)
+            .flatTap(output => cacheDB.put(store)(pickledInput, Pickled(asBytes(output))).toF)
+        )(pickledOutput => F.pure(fromBytes[O](pickledOutput.value)).rethrow)
+      )
+    // .flatTap(o => F.delay(println(s"CACHED RESULT $o")))
+  }
+
+/**
+ * Cache backed up by a JS `Cache`.
+ */
+case class JsCacheCache[F[_]: PromiseConverter](cache: JsCache)(using F: Sync[F]) extends Cache[F]:
+  def eval[I: Pickler, O: Pickler](computation: Cacheable[F, I, O]): I => F[O] = { input =>
+    val pickledInput: Array[Byte] = asBytes(input)
+    val base64Input: String       = Base64.getEncoder.encode(pickledInput).map(_.toChar).mkString
+    val stringInput: String       =
+      s"http://request/${computation.name}/v${computation.version}?$base64Input"
+
+    PromiseConverter[F]
+      .convert(cache.`match`(stringInput))
+      // .flatTap(pickled => F.delay(println(pickled.fold("CACHE MISS")(_ => "CACHE HIT"))))
+      .flatMap(
+        _.fold(
+          computation
+            .invoke(input)
+            .flatTap(output =>
+              PromiseConverter[F].convert(
+                cache.put(stringInput, new dom.Response(asTypedArray(output)))
+              )
+            )
+        )(response =>
+          PromiseConverter[F]
+            .convert(response.arrayBuffer())
+            .flatMap(ab => F.pure(fromTypedArray[O](new Int8Array(ab))).rethrow)
+        )
+      )
+  }
+
+object Cache:
+  def withIDB[F[_]: Async](dbFactory: Option[IDBFactory], dbName: String): F[Cache[F]] =
```

Comment:
Will each worker use its own db, or do they share a db and only differentiate with the `name` space?